windows/Dockerfile

```diff
@@ -27,6 +27,7 @@ ARG GIT_VERSION
 ARG SEVENZIP_VERSION
 ARG VS2017BUILDTOOLS_VERSION
 ARG GO_VERSION
+ARG GO_SHA256_WINDOWS_AMD64
```

Comment:
According to what we do on other Dockerfile I think you would need to add something like `ENV GO_SHA256_WINDOWS_AMD64=$GO_SHA256_WINDOWS_AMD64` to have it as an environment variable that you can use in the ps scripts that are called in the Dockerfile.

I am not sure this is absolutely required tho, if it works like that feel free to ignore the comment

Reply:
The main difference between ENV and ARG is that ARG is build time only while ENV is still accessible at run time.
Both declare environment variables, what change is just when it is accessible, in particular they are both accessible when running a script during the build of the `Dockerfile`.

The following is just a workaround so that an argument persist at run time:
```
ARG name=default
ENV name $name
```

We don't need `GO_SHA256_WINDOWS_AMD64` to be accessible at run time, so no need for an `ENV`.

See https://vsupalov.com/docker-arg-env-variable-guide for a very interesting description of the difference.

Reply:
Oh that's good to know, we could probably clean some of the Dockerfile then. Thanks for the explanation and the link!

Reply:
Yes I thought so too :)